Sockets/Sockets.Implementation.NET/TcpSocketClient.cs

```diff
@@ -55,7 +87,14 @@ internal TcpSocketClient(TcpClient backingClient)
         /// </summary>
         public Stream ReadStream
         {
-            get { return _backingTcpClient.GetStream(); }
+            get
+            {
+                if (_secureStream != null)
```

Comment:
because of this check, we should add `_secureStream = null;` to the `DisconnectAsync` method, in case the user attempts to connect once with secure, and then a second time without secure. Otherwise the correct stream would be hidden by the old _secureStream.